src/Hooks.php

```diff
@@ -86,6 +86,8 @@ public function dispatch($hook, $parameters = []) {
 			$parameters = array_values($parameters);
 		}
 
+		ksort($this->hooks[$hook]);
```

Comment:
I don't think this is stable for hooks with the same priority for PHP < 8.0. So, for example for the below tests, hooks `10-1` and `10-2` are not guaranteed to always stay in that same order after `ksort()`, as the behavior for same keys is undefined. From the PHP manual at https://www.php.net/manual/en/function.ksort.php:
> Note:
>
>If two members compare as equal, they retain their original order. Prior to PHP 8.0.0, their relative order in the sorted array was undefined.

Here's how I solved this in a different project: https://github.com/ampproject/amp-toolbox-php/blob/b7377775da193e9cd93fb904421181b47134af70/src/Validator/ValidationErrorCollection.php#L61-L83

I think we'll need something like this here as well, if we want to guarantee that hooks inserted in a certain order for a same priority stay in that same order.

Reply:
The thing is, there never was any mention (documentation) or guarantee that hooks at the _same_ priority would run in a certain order. So yes, while this _may_ change behaviour for callbacks registered on the _same_ priority, the order for those was never fixed anyway.

What this fixes is that the _`$priority`_ parameter - which was the documented way to influence the run order of callbacks, wasn't being respected and now will be.

So if people want to use a specific run order of their callbacks, they should use the (documented) `$priority` parameter and not rely on registration order.

Reply:
@schlessera See: https://3v4l.org/gAui9 We're not ordering the callbacks, we're ordering the priorities. That's a different level in the array.